### ISSUE TITLE
lint: suggest strings.Builter instead of bytes.Buffer, where applicable

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#630]
Detects `bytes.Buffer` usages that could benefit from `strings.Builder` replacement.

**Before:**
```go
var b bytes.Buffer
b.WriteString(s)
return b.String()
```
**After:**
```go
var b strings.Buffer
b.WriteString(s)
return b.String()
```

Notes: do not trigger on buffers that call to `Grow` or created with `bytes.NewBuffer`.
